R/http.R

```diff
@@ -0,0 +1,106 @@
+#' @title Travis CI HTTP Requests
+#'
+#' @description This is the workhorse function for executing API requests for
+#'   Travis CI.
+#'
+#' @import httr
+#' @importFrom jsonlite fromJSON
+#'
+#' @details This is mostly an internal function for executing API requests. In
+#'   almost all cases, users do not need to access this directly.
+#'
+#' @param verb A character string containing an HTTP verb, defaulting to `GET`.
+#' @param path A character string with the API endpoint (should begin with a
+#'   slash).
+#' @param query A list specifying any query string arguments to pass to the API.
+#'   This is used to pass the API token.
+#' @param body A named list or character string of what should be passed in the
+#'   request. Corresponds to the "-d" argument of the `curl` command.
+#' @template endpoint
+#' @param encode Encoding format. See [httr::POST].
+#'
+#' @return The JSON response, or the relevant error.
+#'
+#' @export
+travisHTTP <- function(verb = "GET",
```

Comment:
Maybe call it `travis()`, in the spirit of `gh::gh()` ?

Reply:
ok

Reply:
Reverted back to `travisHTTP` because we get conflicts with the `travis-package.Rd` file. I am not exporting the function anyways to keep everything tidy so this won't matter.